client/src/main/java/com/vesoft/nebula/client/graph/net/NebulaPool.java

```diff
@@ -106,6 +106,10 @@ public boolean init(List<HostAddress> addresses, NebulaPoolConfig config)
         objConfig.setMinIdle(config.getMinConnSize());
         objConfig.setMaxIdle(config.getMaxConnSize());
         objConfig.setMaxTotal(config.getMaxConnSize());
+        objConfig.setTestOnBorrow(true);
+        objConfig.setTestOnCreate(true);
+        objConfig.setTestOnReturn(true);
+        objConfig.setTestWhileIdle(true);
```

Comment:
What do these parameters do for the `pool`?

Reply:
As in the commons-pool.jar notes, as follows: 
```
OnBorrow
     * Sets whether objects borrowed from the pool will be validated before
     * being returned from the <code>borrowObject()</code> method. Validation is
     * performed by the <code>validateObject()</code> method of the factory
     * associated with the pool. If the object fails to validate, it will be
     * removed from the pool and destroyed, and a new attempt will be made to
     * borrow an object from the pool.
OnCreate
     * Sets whether objects created for the pool will be validated before
     * being returned from the <code>borrowObject()</code> method. Validation is
     * performed by the <code>validateObject()</code> method of the factory
     * associated with the pool. If the object fails to validate, then
     * <code>borrowObject()</code> will fail.
OnReturn
     * Sets whether objects borrowed from the pool will be validated when
     * they are returned to the pool via the <code>returnObject()</code> method.
     * Validation is performed by the <code>validateObject()</code> method of
     * the factory associated with the pool. Returning objects that fail validation
     * are destroyed rather then being returned the pool.
WhileIdle
     * Returns whether objects sitting idle in the pool will be validated by the
     * idle object evictor (if any - see
     * {@link #setTimeBetweenEvictionRunsMillis(long)}). Validation is performed
     * by the <code>validateObject()</code> method of the factory associated
     * with the pool. If the object fails to validate, it will be removed from
     * the pool and destroyed.  Note that setting this property has no effect
     * unless the idle object evictor is enabled by setting
     * <code>timeBetweenEvictionRunsMillis</code> to a positive value.
```

Reply:
I do not know Do I explain your question clearly?

Reply:
@CPWstatic pinging you :)